services/azure-storage/src/provide_credential/azure_pipelines.rs

```diff
@@ -233,12 +233,12 @@ impl ProvideCredential for AzurePipelinesCredentialProvider {
             .await?;
 
         // Calculate expiration time
-        let expires_in = std::time::Duration::from_secs(token_response.expires_in);
-        let expires_on = now().checked_add(expires_in).ok();
+        let expires_in = Duration::from_secs(token_response.expires_in);
+        let expires_on = Timestamp::now() + expires_in;
```

Comment:
We use add almost everywhere. So I'd use add first and leave whether to use add or checked_add a follow-up.